sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -13,13 +17,489 @@ namespace Azure.Iot.Hub.Service
     /// </summary>
     public class DevicesClient
     {
+        private const string ContinuationTokenHeader = "x-ms-continuation";
+
+        private readonly RegistryManagerRestClient _registryManagerClient;
+        private readonly TwinRestClient _twinClient;
+        private readonly DeviceMethodRestClient _deviceMethodClient;
+
+        protected DevicesClient()
+        {
+        }
+
+        internal DevicesClient(RegistryManagerRestClient registryManagerClient, TwinRestClient twinRestClient, DeviceMethodRestClient deviceMethodRestClient)
+        {
+            Argument.AssertNotNull(registryManagerClient, nameof(registryManagerClient));
+            Argument.AssertNotNull(twinRestClient, nameof(twinRestClient));
+            Argument.AssertNotNull(deviceMethodRestClient, nameof(deviceMethodRestClient));
+
+            _registryManagerClient = registryManagerClient;
+            _twinClient = twinRestClient;
+            _deviceMethodClient = deviceMethodRestClient;
+        }
+
+        /// <summary>
+        /// Create or update a device identity.
+        /// </summary>
+        /// <param name="deviceIdentity">The device to create.</param>
+        /// <param name="precondition">The condition on which to perform this operation. To create a device identity, this value must be equal to <see cref="IfMatchPrecondition.Unconditional"/>.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The created device.</returns>
+        public virtual async Task<Response<DeviceIdentity>> CreateOrUpdateIdentityAsync(DeviceIdentity deviceIdentity, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(deviceIdentity, nameof(deviceIdentity));
+            String ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, deviceIdentity.Etag);
+            return await _registryManagerClient.CreateOrUpdateDeviceAsync(deviceIdentity.DeviceId, deviceIdentity, ifMatchHeaderValue, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Create or update a device identity.
+        /// </summary>
+        /// <param name="deviceIdentity">The device to create.</param>
```

Comment:
if we use "identity" suffix in the summary, I think the parameter description also has to include the "identity" keyword